yarn/datadog_checks/yarn/yarn.py

```diff
@@ -144,6 +153,16 @@ class YarnCheck(AgentCheck):
 
     _ALLOWED_APPLICATION_TAGS = ['applicationTags', 'applicationType', 'name', 'queue', 'user']
 
+    def __init__(self, *args, **kwargs):
+        super(YarnCheck, self).__init__(*args, **kwargs)
+        application_status_mapping = self.instances[0].get('application_status_mapping', {})
+        try:
+            self.application_status_mapping = {
+                k.upper(): getattr(AgentCheck, v.upper()) for k, v in application_status_mapping.items()
+            } or DEFAULT_APPLICATION_STATUS_MAPPING
```

Comment:
1. Semantically it makes more sense to add the default here:

```
application_status_mapping = self.instances[0].get('application_status_mapping', DEFAULT_APPLICATION_STATUS_MAPPING)
```

With the current implementation, if the user put an empty mapping (I know, very realistic), we are going to override it with the default mapping.

2. Could you add a test for the default mapping ? :)